src/capwap_loc_provider.erl

```diff
@@ -0,0 +1,138 @@
+%% Copyright (C) 2013-2023, Travelping GmbH <info@travelping.com>
+
+%% This program is free software: you can redistribute it and/or modify
+%% it under the terms of the GNU Affero General Public License as published by
+%% the Free Software Foundation, either version 3 of the License, or
+%% (at your option) any later version.
+
+%% This program is distributed in the hope that it will be useful,
+%% but WITHOUT ANY WARRANTY; without even the implied warranty of
+%% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+%% GNU Affero General Public License for more details.
+
+%% You should have received a copy of the GNU Affero General Public License
+%% along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+-module(capwap_loc_provider).
+
+-behavior(gen_server).
+
+%% API
+-export([start/1, start_link/1]).
+
+%% gen_server callbacks
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2]).
+
+%% user api
+-export([flush_loc_cache/0, get_loc/1, get_loc/2, load_config/1]).
+
+-include_lib("kernel/include/logger.hrl").
+
+-define(CACHE_LOC, capwap_loc_provider_cache).
+-define(CACHE_CONFIG, capwap_loc_config_cache).
```

Comment:
Where this macros is used?

Reply:
This one may not be used. I'll remove it.